src/test/FuzzerImpl.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "simulation/Simulation.h"
+#include "test/Fuzzer.h"
+#include "test/TestAccount.h"
+
+using namespace stellar;
+
+namespace stellar
+{
+
+class TransactionFuzzer : public Fuzzer
+{
+  public:
+    TransactionFuzzer(int numAccounts, int processID);
+    void inject(stellar::XDRInputFileStream&) override;
+    static void generateFuzzerInput(const std::string& filename);
```

Comment:
`generateFuzzerInput` and `isBadInput`should be non-member non-friend functions for better encapsulation - just make them free functions inside of a namespace, e.g. FuzzUtils. (by the way, you can read up here on why this is a better practice: http://www.drdobbs.com/cpp/how-non-member-functions-improve-encapsu/184401197)

Reply:
Thanks for the link!